Packages/com.unity.inputsystem/InputSystem/Actions/InputActionState.cs

```diff
@@ -3907,6 +4035,128 @@ public enum Flags
+         *
+         * Action0
+         *  - c
+         *  - Ctrl+Shift+c
+         *
+         * Action1
+         *  - Ctrl+c
+         *
+         * This could result in the following control groupings and complexities:
+         *
+         *  ID    Group   Control                     Complexity
+         *  1     1       c(from Ctrl+Shift+c)        3
+         *  2     1       c(from Ctrl+c)              2
+         *  3     1       c(from c)                   1
+         *  4     2       Ctrl(from Ctrl+Shift+c)     3
+         *  5     2       Ctrl(from Ctrl+c)           2
+         *  6     3       Shift(from Ctrl+Shift+c)    3
+         *
+         *
+         * which would result in the following action groupings
+         *
+         *  Group1(c)          Group2(ctrl)         Group3(shift)
+         *  Action0            Action0              Action0
+         *  Action1            Action1
+         *
+         * At runtime, lets say Ctrl+c is pressed. We want Action1 to handle this because it has the
+         * highest priority binding that matches. State monitors will have been sorted by complexity,
+         * so the first binding to perform will be Id 2 above. We will look up the group from that
+         * control, group 1 in this case, and then set the 'lastEventHandledByAction' property on
+         * action group 1 to the index of the action that owns the binding that fired, and set the
+         * input event to handled so the next binding to match (the c key) knows not to also handle it.
+         *
+         * When event consumption is turned off, and when using the InputAction.WasPerformedThisFrame()
+         * method, it's necessary to check each action group that an action belongs to to see whether that
+         * action group performed in the current frame and if the action that performed is the action
+         * we called WasPerformedThisFrame() on.
+         */
+        internal struct ActionGroup : IDisposable
+        {
+            /// <summary>
+            /// Stores the indexes into the actionStates array for every action that appears in this group.
+            /// </summary>
+            /// <remarks>
+            /// Currently this is just used to support building the action groups, but later can be used
+            /// to allow querying for conflicting actions at editor time or runtime.
+            ///
+            /// This is a pointer instead of a NativeArray because 2019.4 does not support nesting
+            /// NativeArrays and action group instances themselves live inside a NativeArray.
+            /// </remarks>
+            private int* actionIndicies;
+            private int actionCount;
+
+            public bool isCreated { get; }
+
+            public int lastEventHandledByAction { get; set; }
+
+            private int m_Length;
+
+            public ActionGroup(int capacity)
+            {
+                actionIndicies = (int*)UnsafeUtility.Malloc(capacity, 8, Allocator.Persistent);
```

Comment:
Could we add a named variable / static to make this 8 a bit clearer that its alignment (and why 8 chosen)

Reply:
Went one better.